### PR TITLE
dozenal: init at version v12010904

### PR DESCRIPTION
**Pull request by CharlesHD**
###### Motivation for this change
dozenal is a toolsuite for working with base twelve instead of base ten that I use on a daily basis. 

hdate is a library for working with hebraic date, I do not use it but dozenal depends on hdate, so I added an hdate derivation too. The hdate utilities works too, maybe it can be useful to others.

###### Things done
- Added a hdate derivation, which is a dozenal depency
- Added the dozenal derivation
- Added myself in the list of maintainers
- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by CharlesHD** [comment]
@GrahamcOfBorg build dozenal

**Comment by CharlesHD** [comment]
@GrahamcOfBorg build hdate dozenal

File: pkgs/applications/misc/dozenal/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{ stdenv, fetchFromGitHub, ncurses, hdate, lua5_2 }:
+
+stdenv.mkDerivation rec {
+  name = "dozenal";
+  version = "v12010904";
```
Comment (joachifm) [inline pkgs/applications/misc/dozenal/default.nix]: Note that `version` should be a part of `name` and should not contain the `v` prefix.  You can check `builtins.parseDrvName` to see how version strings are parsed out of the derivation name.

**Comment by joachifm** [comment]
Needs a rebase

**Comment by CharlesHD** [comment]
@joachifm thanks for your review. I think I took care of what you pointed out.

**Comment by joachifm** [comment]
@GrahamcOfBorg build hdate dozenal

**Pull request merged**